pubsub/google/cloud/pubsub/topic.py

```diff
@@ -435,12 +462,27 @@ def publish(self, message, **attrs):
 
         :type attrs: dict (string -> string)
         :param attrs: key-value pairs to send as message attributes
+
+        :rtype: None
+        :returns: None
         """
         self.topic._timestamp_message(attrs)
         self.messages.append(
             {'data': message,
              'attributes': attrs})
 
+        # If too much time has elapsed since the first message
+        # was added, autocommit.
+        if self._max_interval < self.INFINITY:
+            if float(time.time()) - self._start_timestamp > self._max_interval:
+                self._start_timestamp = float(time.time())
```

Comment:
1.  Why cast to `float()`? (as above)
1.  Don't grab `time.time()` twice like this? Instead

    ```python
    if self._max_interval < INFINITY:
        now = time.time()
        if now - self._start_timestamp > self._max_interval:
            self.commit()
            self._start_timestamp = now
            return
    ```

1. Are you worried about setting `self._start_timestamp` and then `self.commit()` failing? i.e. does that put the instance in a bad state?